### DISCUSSION TITLE
Question abouts Solids Reactivity

### DISCUSSION THREAD

**Discussion by reramjiawan** [solidjs/solid#1002] [Q&A] [answered]
From my understanding Solid should only update what is affected. But after playing around I noticed that Solid also calls functions that are not directly dependent on a signal which can be seen in the following playground:

https://playground.solidjs.com/?version=1.4.1#NobwRAdghgtgpmAXGGUCWEwBowBcCeADgsrgM4Ae2YZA9gK4BOAxiWGjIbY7gAQi9GcCABM4jXgF9eAM0a0YvADo1aAGzQiAtACsyAegDucAEYqA3EogcuPfr2ZCouOAGU0Ac2hqps+YpU6DW09CysrGXoIZlw0WgheAGEGCBdGAAoASn4rXgd4sj5gZhTcLF4yOFxkqNwAXV4AXgcnF3cvKDV0gAZMywhciqqAFQ44Blx0rKaAPhyEvKHq0vSAJj7ByXLV7t3M8IWSiELeDyqASREm3mnGuZBBvKOguAA6NVoPdIByM9xLhydNRwK5QDzoBJQGRpfK1Xj0QgiZxwb4bBZ5IS4JgJb6ab68ADUsNSWX6eUkByeBT4f1cBGB11u90e+WO6jeHy+vyqdPwDM6dEBamBVzgADdhLxcAALBgeaW8NDkXgQWh8MTEUTCXCosmLTHY+aLRYmKDMADWHnkUREiF43yEIm+LIpC1dgwNjAS6RZAB4TPRcLh4oqRI0QH9LllpIU+XBw7T6XBozMWXkQCVatG-foA0H4qmFmj3RAhFqMkzeL6aqlxLx9DNyiJaMx6PBUq8-gBRYHt3AAIXwUZUUEIhBUmQAhGiwJI6kA

I was under the assumption Solid knows what to rerun and what not. 

Kind regards.

**Comment by ryansolid** [comment, marked as answer]
Yeah to reduce creation cost we do group expressions together in the same template that aren't heavier inserts. The cost of doing a quick shallow referential check is usually much cheaper than creating independent reactive computations, so it scales better. By isolating anything that would require DOM diffing we still get performance gains. If the thing is expensive you could wrap it in a `createMemo`,  but most of the time that isn't necessary.

Your element compiles to something similar to this if you look at the output tab:
```js
createEffect((previous = {}) => {
  const id = getId();
  const style = getStyle();

  id !== previous.id && setAttribute(_el$, "id", previous.id = id);
  previous.style = setStyle(_el$, style, previous.style);
  return previous;
});
```

I talk about it a bit here. But the secret to Solid's performance isn't just that it's fine-grained but that we can have control over the granularity of change.

https://dev.to/ryansolid/thinking-granular-how-is-solidjs-so-performant-4g37
